services/consensusalgo/benchmarkconsensus/test/harness.go

```diff
@@ -71,3 +76,82 @@ func (h *harness) createService(ctx context.Context) {
 		h.config,
 	)
 }
+
+func nonLeaderPublicKey() primitives.Ed25519Pkey {
+	return []byte{0x99}
+}
+
+func (h *harness) expectHandlerRegistrations() {
+	h.gossip.Reset().When("RegisterBenchmarkConsensusHandler", mock.Any).Return().Times(1)
+	h.blockStorage.Reset().When("RegisterConsensusBlocksHandler", mock.Any).Return().Times(1)
+}
+
+func (h *harness) verifyHandlerRegistrations(t *testing.T) {
+	ok, err := h.gossip.Verify()
+	if !ok {
+		t.Fatal("Did not register with Gossip:", err)
+	}
+	ok, err = h.blockStorage.Verify()
+	if !ok {
+		t.Fatal("Did not register with BlockStorage:", err)
+	}
+}
+
+func (h *harness) expectBlockCreation() {
+	h.consensusContext.Reset().When("RequestNewTransactionsBlock", mock.Any).Return(nil, nil).AtLeast(1)
```

Comment:
the name of the function doesn't reflect what it does. It doesn't actually expect that a block was created, right? it expects the algo to ask for a new transaction block... I'd name it `expectRequestNewTransactionBlock()`.

in addition, maybe return the `MockFunction` so that I could do:
```golang
h.expectRequestNewTransactionBlock().AtLeast(1)
h.expectRequestNewTransactionBlock().Times(0)

// etc
```

Reply:
Good idea